tests/e2e/plugins/user-input.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Plugin Name: E2E Tests User Input Settings API Mock
+ * Plugin URI:  https://github.com/google/site-kit-wp
+ * Description: Utility plugin for handling user input settings for Site Kit during E2E tests.
+ * Author:      Google
+ * Author URI:  https://opensource.google.com
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Google_Proxy;
+use Google\Site_Kit\Core\Authentication\User_Input_State;
+use Google\Site_Kit\Core\Storage\User_Options;
+
+register_activation_hook( __FILE__, 'e2e_user_input_settings_reset' );
+register_deactivation_hook( __FILE__, 'e2e_user_input_settings_reset' );
+
+add_filter( 'pre_http_request', 'e2e_user_input_settings_proxy_handler', 10, 3 );
+
+function e2e_user_input_settings_reset() {
+	$user_options     = new User_Options( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+	$user_input_state = new User_Input_State( $user_options );
+
+	$user_input_state->delete();
+
+	delete_option( '_userinput_sitewide' );
+	delete_option( '_userinput_' . get_current_user_id() );
+}
+
+function e2e_user_input_settings_proxy_handler( $pre, $args, $url ) {
+	$google_proxy            = new Google_Proxy( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+	$user_input_settings_url = $google_proxy->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
+	if ( $url !== $user_input_settings_url ) {
+		return $pre;
+	}
+
+	$user_id  = get_current_user_id();
+	$defaults = array(
```

Comment:
Is all of this necessary/used by the tests? It looks like most of this might have been copied from before so I realize it probably wasn't all written for this, but it would be good to keep this to only what's needed if we can reduce it. What do you think?

Reply:
It's needed because that's what we expect to receive even if there is nothing saved yet. We can't remove it.